src/Symfony/Component/HttpFoundation/RequestMatcher.php

```diff
@@ -11,6 +11,8 @@
 
 namespace Symfony\Component\HttpFoundation;
 
+trigger_deprecation('symfony/http-foundation', '6.2', 'The "%s" class is deprecated, use "%s" instead.', RequestMatcher::class, ChainRequestMatcher::class);
```

Comment:
oh, so we deprecate the current API, OK then.
`@deprecated` annotations are missing